sunpy/net/tests/test_helioviewer.py

```diff
@@ -88,16 +89,17 @@ def test_get_jp2_header(self, client):
         assert ('fits' in header1.keys()) and ('fits' in header2.keys())
 
     @skip_glymur
-    def test_download_jp2_map(self, client):
+    @figure_test
+    def test_download_jp2_map(self, client, tmpdir):
         """
-        Tests getJP2Image API method with Map.
+        Tests getJP2Image API method with Map with a figure test.
         """
-        # TODO: make this a figure test.
         filepath = client.download_jp2('2012/01/01', observatory='SOHO',
-                                       instrument='MDI', measurement='continuum')
+                                       instrument='MDI', measurement='continuum',
+                                       directory=tmpdir.dirname)
         map_ = sunpy.map.Map(filepath)
-        assert isinstance(map_, sunpy.map.GenericMap)
-        os.remove(filepath)
+        with pytest.raises(SunpyMetadataWarning, match="Missing metadata for observer: assuming Earth-based observer."):
+            map_.peek()
```

Comment:
Probably shouldn't use `peek()` to generate the figure, but we don't need to open that can of worms here